XmlFormatter/src/Settings/SettingsManager.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System.Collections.Generic;
+using XmlFormatter.src.Interfaces.Settings;
+using XmlFormatter.src.Interfaces.Settings.DataStructure;
+using XmlFormatter.src.Interfaces.Settings.LoadingProvider;
+
+namespace XmlFormatter.src.Settings
+{
+    /// <summary>
+    /// This class is the default settings manager
+    /// </summary>
+    class SettingsManager : ISettingsManager
+    {
+        /// <summary>
+        /// All the scopes in this manager
+        /// </summary>
+        private readonly List<ISettingScope> scopes;
+
+        /// <summary>
+        /// The factory to use for getting the saving and loading provider
+        /// </summary>
+        private IPersistentFactory persistentFactory;
+
+        /// <summary>
+        /// The current loading provider to use
+        /// </summary>
+        private ISettingLoadProvider loadProvider;
+
+        /// <summary>
+        /// The current saving provider to use
+        /// </summary>
+        private ISettingSaveProvider saveProvider;
+
+        /// <summary>
+        /// Create a new empty instance of this manager
+        /// </summary>
+        public SettingsManager()
+        {
+            scopes = new List<ISettingScope>();
+        }
+
+        /// <inheritdoc/>
+        public void SetPersistendFactory(IPersistentFactory factory)
+        {
+            persistentFactory = factory;
+            loadProvider = null;
+            saveProvider = null;
+        }
+
+        /// <inheritdoc/>
+        public void AddScope(ISettingScope newScope)
+        {
+            scopes.RemoveAll((currentScope) =>
+            {
+                return currentScope.Name == newScope.Name;
+            });
+            scopes.Add(newScope);
+        }
+
+        /// <inheritdoc/>
+        public ISettingScope GetScope(string name)
+        {
+            return scopes.Find((currentScope) =>
+            {
+                return currentScope.Name == name;
+            });
+        }
+
+        /// <inheritdoc/>
+        public List<ISettingScope> GetScopes()
+        {
+            return scopes;
+        }
+
+        /// <inheritdoc/>
+        public bool Load(string filePath)
+        {
+            if (persistentFactory == null)
+            {
+                return false;
+            }
+            loadProvider = loadProvider ?? persistentFactory.CreateLoader();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '??=' operator here.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5053335)